Docker/README.md

```diff
@@ -0,0 +1,150 @@
+### Introduction
+
+This directory contains all the necessary configuration to allow someone
+quickly setup and spin-up a small Narwhal cluster via [docker-compose](https://docs.docker.com/compose/).
+
+Under this directory there will be found the following 2 things:
+* The `Docker` file definition for a Narwhal node
+* A `docker-compose` file to allow someone quickly spin-up a Narwhal cluster
+
+### Quick start
+
+The following dependencies must be installed before continuing further.
+
+#### Docker
+Please find installation info [here](https://docs.docker.com/get-docker/)
+
+#### Docker-compose
+Please find installation info [here](https://docs.docker.com/compose/install/)
+
+After having installed `Docker` & `docker-compose`, next step will be to
+start the cluster via the following command:
+```
+docker-compose -f docker-compose.yml up
+```
+The first time this will run, it will build the narwhal docker image (this can take a few minutes
+since the narwhal node binary needs to be built from the source code) and then it will spin up 
+a cluster for `4 nodes` by doing the necessary setup for `primary` and `worker` nodes. Each
+`primary` node will be connected to `1 worker` node.
+
+The logs for each authority (primary & worker) can be found on the `logs` folder under the corresponding
+authority folder. The `logs` folder will be created once the node is bootstrapped via docker-compose. 
+For example, for the primary node of the authority-0, the logs will be found under
+the folder [logs](authorities/authority-0/logs) and with the name `log-primary.txt`. To monitor the logging
+of a node in real time you can just do something like:
+```
+tail -f authorities/authority-0/logs/log-primary.txt
+```
+
+By default, the development version of the Narwhal node will be compiled when the Docker image is being built.
+To build the Docker image with the production version of it - which will contain all the Rust optimisations,
+you can run the docker-compose command as:
+```
+docker-compose build --build-arg BUILD_MODE=--release
+
+# and then run as
+
+docker-compose up
+```
+
+**Note:** by default each authority's directory will be cleaned up between docker-compose runs when each node
+bootstraps. To preserve those between runs please see the usage of the environment variable `CLEANUP_DISABLED` on
+the [section](#docker-compose-configuration) bellow.
+
+### Access primary node gRPC endpoint
+
+The nodes by default are running with the `Tusk` algorithm disabled, which basically allow
+to user to treat Narwhal as a pure mempool. When that happens, the gRPC server is bootstrapped
+for the primary nodes and that allow someone to interact with the node.
+
+The docker-compose file is exporting for the `primary` nodes the gRPC server's
+port so it can be accessible from the host machine. For the default setup of `4 primary`
+nodes, the `gRPC` ports are the following:
+* `primary_0`: 8000
+* `primary_1`: 8001
+* `primary_2`: 8002
+* `primary_3`: 8003
```

Comment:
We should be clear these are local ports, and be clear that they are the exposition of the same inner port (8000), pointing the reader to look into the docker-compose file for details. 

The reason why this is the case is:
1. someone is bound to ask why the ports are different,
2. you'd want to change this config when running docker-compose to deploy [on say, ecs](https://aws.amazon.com/blogs/containers/deploy-applications-on-amazon-ecs-using-docker-compose/)

Reply:
Good point, I'll do that!